### PR TITLE
Raw Mesh3d feature

### PR DESCRIPTION
**Pull request by dabingnn**
Do not merge it please.
Hi
Here is the raw version of our sprite3d. There could be some bugs and faults in the code, and we are continue to work on it.
This pull request is opened to get comments, reviews together with feedbacks.
Any suggestions and comments is appreciated.
Thanks.

### CODE REVIEW

**Comment by ricardoquesada** [comment]
[ci build]

**Comment by ricardoquesada** [comment]
[ci rebuild]

**Comment by dabingnn** [comment]
[ci rebuild]

**Comment by dumganhar** [comment]
CI build failed. Could you check it? It seems to be the problem of lua bindings.

**Comment by dabingnn** [comment]
[ci rebuild]

**Comment by dabingnn** [comment]
We have achieved a milestone.
And the code is ready for review and merge.
Thanks

**Comment by dumganhar** [comment]
@dabingnn , I fixed the lua bindings generated issues in the latest codes, please pull the latest codes on v3 branch. Thanks.

File: cocos/3d/CCMesh.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+{
+    
+}
+
+Mesh::~Mesh()
+{
+    cleanAndFreeBuffers();
+}
+
+Mesh* Mesh::create(std::vector<float>& posions, std::vector<float>& normals, std::vector<float>& texs, const std::vector<unsigned short>& indices)
```
Comment (dumganhar) [inline cocos/3d/CCMesh.cpp]: 1) Typo: posions -> positions
2) Please use `const &`.

**Comment by dabingnn** [comment]
Thanks James.
Could you please help us check the build status? It always failed.
But all the platforms work correctly when testing it.

**Comment by dumganhar** [comment]
OK, It seems that you didn't change the configuration file for luabindings. I will check it.

File: cocos/3d/CCMesh.cpp (hunk 1 above)
Comment (dabingnn) [inline cocos/3d/CCMesh.cpp]: OK
Thanks.

**Comment by dumganhar** [comment]
PR done: https://github.com/super626/cocos2d-x/pull/34
Please merge it.

**Comment by dumganhar** [comment]
[ci rebuild]

**Comment by ricardoquesada** [comment]
I'll test this, create a "squashed" PR for this, and then I'll merge it.

**Comment by ricardoquesada** [comment]
PR merged here:
https://github.com/cocos2d/cocos2d-x/pull/6794
(squashed).

Closing this PR.

**Pull request closed without merging**